src/main/java/org/reactome/addlinks/referencecreators/EnsemblBioMartMicroarrayPopulator.java

```diff
@@ -0,0 +1,159 @@
+package org.reactome.addlinks.referencecreators;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.reactome.addlinks.EnsemblBioMartUtil;
+
+import java.util.*;
+
+public class EnsemblBioMartMicroarrayPopulator extends SimpleReferenceCreator <Map<String, List<String>>>{
+
+    public EnsemblBioMartMicroarrayPopulator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB)
+    {
+        super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB);
+    }
+
+    public EnsemblBioMartMicroarrayPopulator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB, String refCreatorName)
+    {
+        super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB, refCreatorName);
+    }
+
+    /**
+     * Adds microarray data to the 'otherIdentifier' attribute of ReferenceGeneProduct (RGP) instances. This class doesn't create
+     * any cross-references for the microarray data, since URLs do not exist for them. Reactome's analysis tool still requires the
+     * microarray data though, so this class populates it.
+     * @param personID - Newly created Identifiers will be associated with an InstanceEdit. That InstanceEdit will be associated with the Person entity whose ID == personID
+     * @param mappings Identifier mapping structure that contains all data used to populate microarray data to RGPs.
+     * @param sourceReferences - A list of GKInstance objects that were in the database. References will be created if they are in the keys of mapping and also in sourceReferences.
+     * This is to handle cases where a mapping comes from a third-party file that may contain source identifiers that are in *their* system but aren't actually in Reactome.
+     * @throws Exception - Thrown by the MySQLAdaptor/GKInstance classes. IOException/ParseException can also be thrown when getting species names from Species.json file.
+     */
+    @Override
+    public void createIdentifiers(long personID, Map<String, Map<String, List<String>>> mappings, List<GKInstance> sourceReferences) throws Exception {
+
+        // Iterate through each species that we add microarray data to.
+        for (String speciesName : EnsemblBioMartUtil.getSpeciesNames()) {
+            logger.info("Populating microarray data for " + speciesName);
+            String speciesBiomartName = EnsemblBioMartUtil.getBioMartSpeciesName(speciesName);
+
+            // Retrieve identifier mappings that are used from the 'super' mapping.
+            // TODO: Helper class for getting species data
```

Comment:
I have an idea for what the helper class could look like if you'd like to go over it together.

Reply:
This might be an artifact now, I think the helper method is the `getThingToOtherThingMappings` methods that are below